src/OpenTok/Util/Client.php

```diff
@@ -30,41 +35,71 @@
 /**
 * @internal
 */
-class Client extends \Guzzle\Http\Client
+class Client
 {
     protected $apiKey;
     protected $apiSecret;
     protected $configured = false;
+    protected $client;
 
-    public function configure($apiKey, $apiSecret, $apiUrl)
+    public function configure($apiKey, $apiSecret, $apiUrl, $options = array())
     {
         $this->apiKey = $apiKey;
         $this->apiSecret = $apiSecret;
-        $this->setBaseUrl($apiUrl);
-        $this->setUserAgent(OPENTOK_SDK_USER_AGENT, true);
 
-        // TODO: attach plugins
-        $opentokAuthPlugin = new Plugin\OpentokAuth($apiKey, $apiSecret);
-        $this->addSubscriber($opentokAuthPlugin);
+        if (empty($options['handler'])) {
+            $handlerStack = HandlerStack::create();
+        } else {
+            $handlerStack = $options['handler'];
+        }
+
+        $handler = Middleware::mapRequest(function (RequestInterface $request) {
+            $authHeader = $this->createAuthHeader();
+            return $request->withHeader('X-OPENTOK-AUTH', $authHeader);
+        });
+        $handlerStack->push($handler);
+
+        $ua = OPENTOK_SDK_USER_AGENT . ' ' . \GuzzleHttp\default_user_agent();
+        $this->client = new \GuzzleHttp\Client([
+            'base_uri' => $apiUrl,
+            'handler' => $handlerStack,
+            'headers' => [
+                'User-Agent' => $ua
+            ]
+        ]);
 
         $this->configured = true;
     }
 
-    public function isConfigured() {
+    public function isConfigured()
+    {
         return $this->configured;
     }
 
+    private function createAuthHeader()
+    {
+        $token = array(
+            'ist' => 'project',
+            'iss' => $this->apiKey,
+            'iat' => time(), // this is in seconds
+            'exp' => time()+(5 * 60),
+            'jti' => uniqid(),
+        );
+        return JWT::encode($token, $this->apiSecret);
+    }
+
     // General API Requests
 
     public function createSession($options)
     {
-        $request = $this->post('/session/create');
-        $request->addPostFields($this->postFieldsForOptions($options));
+        $request = new Request('POST', '/session/create');
         try {
-            $sessionXml = $request->send()->xml();
+            $response = $this->client->send($request, [
```

Comment:
Didn’t take the opportunity to go to the JSON-based session create endpoint?

Reply:
Well observed :) I thought I should keep this PR purely for the guzzle change and rather than complicate the changes even more than they already are. I'll do this next, should be simple, just change the request Content-Type header and replace the xml parsing.
Thanks @aoberoi! Hope you're doing well :)